### PR TITLE
Added a spec and tests for filtering by project

### PR DESCRIPTION
**Pull request by adsantos**
This PR introduces a change to the input data: the `indent` field in each project. This is required in order to know which projects are subproject of a given project.
Another particularity on this filter is that the white spaces in the project name cannot be ignored, since the user can create a project with white spaces, e.g. "Doist Work".

### CODE REVIEW

**Comment by adsantos** [comment]
Ready for review

File: input_project.json [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+            "id": 6,
+            "content": "Buy butter.",
+            "dueDate": null,
+            "priority": 4,
+            "order": 5,
+            "assignedBy": {
+                "fullName": "Mary Jones",
+                "email": "mary@doist.io"
+            },
+            "responsible": null,
```
Comment (Garee) [inline input_project.json]: If `assignedBy

Comment (Garee) [inline input_project.json]: If `assignedBy` is populated we should also populate `responsible` as the person the item was assigned to.

File: input_project.json [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+            },
+            "project": 5
+        },
+        {
+            "id": 8,
+            "content": "Buy jeans",
+            "dueDate": "2015-08-11T18:25:43.511Z",
+            "priority": 4,
+            "order": 7,
+            "assignedBy": null,
```
Comment (Garee) [inline input_project.json]: If `responsible

Comment (Garee) [inline input_project.json]: If `responsible` is populated we should populate who it was `assignedBy`.

File: spec_project.json [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+                            "labels": null
+                        }
+                    ]
+                },
+                {
+                    "id": "5",
+                    "freezeDate" : null,
+                    "currentUser" : null,
+                    "context": "When asked for tasks in the project 'Doist' (partial name of the project) in input_project.json",
+                    "query": "p:Doist work",
```
Comment (Garee) [inline spec_project.json]: `query` needs updated to match `context` i.e. set to `Doist`.

File: spec_project.json [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+                    "testFile": "input_project.json",
+                    "assertSections": [
+                        {
+                            "name": "Doist work",
```
Comment (Garee) [inline spec_project.json]: `name` needs updated to `Doist`.

File: spec_project.json [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+                            "labels": null
+                        }
+                    ]
+                },
+                {
+                    "id": "6",
+                    "freezeDate" : null,
+                    "currentUser" : null,
+                    "context": "When asked for tasks in the project 'Wor' (partial name of the project) in input_project.json",
+                    "query": "p:Doist work",
```
Comment (Garee) [inline spec_project.json]: `query` needs updated to match `context` i.e. set to `Wor`.

File: spec_project.json [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+                    "testFile": "input_project.json",
+                    "assertSections": [
+                        {
+                            "name": "Doist work",
```
Comment (Garee) [inline spec_project.json]: `name` needs updated to `Wor`.

File: spec_project.json (hunk 4 above)
Comment (adsantos) [inline spec_project.json]: I think the name should still be "Doist work" because the section name is the project name and not the query. Do you agree?

**Comment by adsantos** [comment]
@Garee It's all updated except the project name, as explained in the previous comment.

File: spec_project.json (hunk 4 above)
Comment (Garee) [inline spec_project.json]: Yes I agree, I was thinking of the current implementation but this is an improvement.

**Pull request merged**